# -*- coding: utf-8 -*-
"""pvalue_prac.ipynb

Automatically generated by Colaboratory.

Original file is located at
    https://colab.research.google.com/drive/1e__eI6HzpQ4WtX6s8Py3_dzffVhkI-3V
"""

import pandas as pd
import numpy as np
from statsmodels.stats.weightstats import ttest_ind

from scipy import stats

"""**gene expression_normal**"""

aml_normal = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/gene/exp_risk_normal.csv',index_col=0)

aml_normal

exp = pd.read_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/gene/exp_normal_data.csv",index_col=0)
exp

c=0
for x in exp.columns:
  for y in exp.index:
    c+=1
print("count :",c)

sample_1=exp
sample_1

sample_1.to_csv("/content/expT.csv")

sample_2=aml_normal
sample_2

t2,p2 = stats.ttest_ind(sample_1, sample_2)
t2,p2

count = 0
for x in p2:
  if x < 0.001:
    count+=1
print("gene_expression_normal :",count)

df = pd.DataFrame(data={"gene_name": sample_1.columns,"p-value": p2})
df

df = pd.DataFrame(data={"gene_name": sample_1.columns,"p-value": p2})
df.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/gene/gene_pValue.csv",index=False)

data = pd.read_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/gene/gene_pValue.csv")
data

data['p-value'] < 0.0001

pd.DataFrame(data['gene_name'] and data['p-value'] < 0.00000000000000000000001)

gene_normal = data[data['p-value'] < 0.001]
gene_normal

gene_normal.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/gene/gene_Value_normal.csv",index=False)

"""gene expression_poor"""

exp_poor = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/gene/exp_poor_data.csv',index_col=0)

exp_poor

sample_exp_poor = exp_poor

gene_exp_risk_category_poor = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/gene/exp_risk_poor.csv',index_col=0)

gene_exp_risk_category_poor

sample_3=gene_exp_risk_category_poor
sample_3

t3,p3 = stats.ttest_ind(sample_exp_poor, sample_3)
t3,p3

count = 0
for x in p3:
  if x < 0.001:
    count+=1
print("gene_expression_poor :",count)

df_risk = pd.DataFrame(data={"gene_name": sample_exp_poor.columns,"p-value": p3})
df_risk

df_risk = pd.DataFrame(data={"gene_name": sample_exp_poor.columns,"p-value": p3})
df_risk.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/gene/gene_pValue_risk.csv",index=False)

data_gene_risk = pd.read_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/gene/gene_pValue_risk.csv")
data_gene_risk

gene_risk = data_gene_risk[data_gene_risk['p-value'] < 0.001]
gene_risk

gene_risk

gene_risk.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/gene/gene_Value_risk.csv",index=False)



"""**DNA methylation normal**"""

methy= pd.read_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/methy/methy_normal_data.csv",index_col=0)
methy

sample_1_methy=methy
sample_1_methy

sample_1_methy.to_csv("/content/methyT.csv")

methy_normal = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/methy/methy_risk_normal.csv',index_col=0)

methy_normal

sample_2_methy_normal = methy_normal

t3_methy,p3_methy_normal = stats.ttest_ind(sample_1_methy, sample_2_methy_normal) 
t3_methy,p3_methy_normal

count = 0
for x in p3_methy_normal:
  if x < 0.001:
    count+=1
print("methy_normal :",count)

methy_normal = pd.DataFrame(data={"gene_name": sample_1_methy.columns,"p-value": p3_methy_normal})
methy_normal.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/methy/methy_pValue_normal.csv",index=False)

data_methy_normal = pd.read_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/methy/methy_pValue_normal.csv")
data_methy_normal

methy_normal = data_methy_normal[data_methy_normal['p-value'] < 0.001]
methy_normal

methy_normal.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/methy/methy_Value_normal.csv",index=False)

"""**DNA methylation poor**"""

methypoorDATA = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/methy/methy_poor_data.csv',index_col=0)

methypoorDATA

sample_methypoorDATA= methypoorDATA

methy_risk_category_poor = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/methy/methy_risk_poor.csv',index_col=0)

methy_risk_category_poor

sample_2_methy_poor = methy_risk_category_poor

t3_methy_poor,p3_methy_poor = stats.ttest_ind(sample_methypoorDATA, sample_2_methy_poor) 
t3_methy_poor,p3_methy_poor

count = 0
for x in p3_methy_poor:
  if x < 0.001:
    count+=1
print("methy_poor :",count)

methy_poor = pd.DataFrame(data={"gene_name": sample_methypoorDATA.columns,"p-value": p3_methy_poor})
methy_poor.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/methy/methy_pValue_poor.csv",index=False)

data_methy_poor = pd.read_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/methy/methy_pValue_poor.csv")
data_methy_poor

methy_poor = data_methy_poor[data_methy_poor['p-value'] < 0.001]
methy_poor

methy_poor.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/methy/methy_Value_poor.csv",index=False)

"""**omics1_3 risk normal**"""

omics1_3 = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic1/omics1_3_normal.csv',index_col=0)

omics1_3

sample1_omics1_3 = omics1_3

sample2_omic1_3_normal = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic1/omcs1_3__risk_normal.csv',index_col=0)

t2_omics1_3_normal,p2_omics1_3_normal = stats.ttest_ind(sample1_omics1_3, sample2_omic1_3_normal)
t2_omics1_3_normal,p2_omics1_3_normal

count = 0
for x in p2_omics1_3_normal:
  if x < 0.001:
    count+=1
print("synthetic gene_expression_normal :",count)

omics1_3_normal = pd.DataFrame(data={"gene_name": sample1_omics1_3.columns,"p-value": p2_omics1_3_normal})
omics1_3_normal.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic1/omics1_3_pValue_normal.csv",index=False)

data_omics1_3_normal= pd.read_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic1/omics1_3_pValue_normal.csv")
data_omics1_3_normal

data_omics1_3_normal_geneName = data_omics1_3_normal[data_omics1_3_normal['p-value'] < 0.001]
data_omics1_3_normal_geneName

data_omics1_3_normal_geneName.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic1/data_omics1_3_normal_geneName.csv",index=False)

"""**omics1_3 poor**"""

sample1_omic1_3_poor = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic1/omics1_3_poor.csv',index_col=0)

sample2_omic1_3_poor = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic1/omics1_3_risk_poor.csv',index_col=0)

t2_omics1_3_poor,p2_omics1_3_poor = stats.ttest_ind(sample1_omic1_3_poor, sample2_omic1_3_poor)
t2_omics1_3_poor,p2_omics1_3_poor

count = 0
for x in p2_omics1_3_poor:
  if x < 0.001:
    count+=1
print("synthetic gene_expression_poor :",count)

omics1_3_poor = pd.DataFrame(data={"gene_name": sample1_omic1_3_poor.columns,"p-value": p2_omics1_3_poor})
omics1_3_poor.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic1/omics1_3_pValue_poor.csv",index=False)

data_omics1_3_poor_geneName = pd.read_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic1/omics1_3_pValue_poor.csv")
data_omics1_3_poor_geneName

data_omics1_3_poor_geneName = data_omics1_3_poor_geneName[data_omics1_3_poor_geneName['p-value'] < 0.001]
data_omics1_3_poor_geneName

#data_omics1_3_normal_geneName.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic1/omics1_3_Value_normal.csv",index=False)

"""**omics2_1 normal**"""

omics2_1 = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic2/omics2_1_normal_data.csv',index_col=0)
omics2_1

sample1_omics2_1 = omics2_1

sample2_omic2_1_Intermediate = pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic2/omics2_1_risk_normal.csv',index_col=0)

sample2_omic2_1_Intermediate

t2_omics2_1_in,p2_omics2_1_in = stats.ttest_ind(sample1_omics2_1, sample2_omic2_1_Intermediate)
t2_omics2_1_in,p2_omics2_1_in

count = 0
for x in p2_omics2_1_in:
  if x < 0.001:
    count+=1
print("synthetic methylation normal :",count)

omic2_1_normal = pd.DataFrame(data={"gene_name": sample1_omics2_1.columns,"p-value": p2_omics2_1_in})
omic2_1_normal.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic2/omic2_1_normal.csv",index=False)

data_omic2_1_normal = pd.read_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic2/omic2_1_normal.csv")
data_omic2_1_normal

data_omic2_1_normal = data_omic2_1_normal[data_omic2_1_normal['p-value'] < 0.001]
data_omic2_1_normal

data_omic2_1_normal.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic2/omic2_1_Value_normal.csv",index=False)

"""**omics2_1 poor**"""

sample1_omic2_1_Poor =  pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic2/omics2_1_poor_data.csv',index_col=0)

sample3_omic2_1_Poor =  pd.read_csv('/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic2/omics2_1_risk_poor.csv',index_col=0)

sample3_omic2_1_Poor

t2_omics2_1_poor,p2_omics2_1_poor = stats.ttest_ind(sample1_omic2_1_Poor, sample3_omic2_1_Poor)
t2_omics2_1_poor,p2_omics2_1_poor

count = 0
for x in p2_omics2_1_poor:
  if x < 0.001:
    count+=1
print("synthetic methylatin poor :",count)

omic2_1_poor = pd.DataFrame(data={"gene_name": sample1_omic2_1_Poor.columns,"p-value": p2_omics2_1_poor})
omic2_1_poor.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic2/omic2_1_poor_P-value.csv",index=False)

data_omic2_1_poor = pd.read_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic2/omic2_1_poor_P-value.csv")
data_omic2_1_poor

data_omic2_1_poor = data_omic2_1_poor[data_omic2_1_poor['p-value'] < 0.001]
data_omic2_1_poor

data_omic2_1_poor.to_csv("/content/drive/MyDrive/omicData/aml_risk_category/gene_methy_separate normal_poor data/omic2/omic2_1_Value_poor.csv",index=False)